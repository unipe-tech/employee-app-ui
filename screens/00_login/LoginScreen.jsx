import { useNavigation } from "@react-navigation/core";
import { useEffect, useState } from "react";
import { Alert, BackHandler, SafeAreaView, Text, View } from "react-native";
import SmsRetriever from "react-native-sms-retriever";
import SplashScreen from "react-native-splash-screen";
import { useDispatch, useSelector } from "react-redux";
import Otp from "../../apis/login/Otp";
import SVGImg from "../../assets/UnipeLogo.svg";
import FormInput from "../../components/atoms/FormInput";
import TermsAndPrivacyModal from "../../components/molecules/TermsAndPrivacyModal";
import { COLORS, FONTS } from "../../constants/Theme";
import { KeyboardAvoidingWrapper } from "../../KeyboardAvoidingWrapper";
import { addPhoneNumber } from "../../store/slices/authSlice";
import {
  addCurrentScreen,
  addCurrentStack,
} from "../../store/slices/navigationSlice";
import { styles } from "../../styles";
import privacyPolicy from "../../templates/docs/PrivacyPolicy.js";
import termsOfUse from "../../templates/docs/TermsOfUse.js";

const LoginScreen = () => {
  SplashScreen.hide();
  const dispatch = useDispatch();
  const [disabled, setDisabled] = useState(true);

  const authSlice = useSelector((state) => state.auth);
  const [phoneNumber, setPhoneNumber] = useState(authSlice?.phoneNumber);

  const [isPrivacyModalVisible, setIsPrivacyModalVisible] = useState(false);
  const [isTermsOfUseModalVisible, setIsTermsOfUseModalVisible] =
    useState(false);

  useEffect(() => {
    dispatch(addCurrentStack("OnboardingStack"));
    dispatch(addCurrentScreen("Login"));
  }, []);

  useEffect(() => {
    dispatch(addPhoneNumber(phoneNumber));
  }, [phoneNumber]);

  useEffect(() => {
    var phoneno = /^[0-9]{10}$/gm;
    if (phoneno.test(phoneNumber) && phoneNumber.length === 10) {
      setDisabled(false);
      console.log("true");
    } else {
      setDisabled(true);
      console.log("false");
    }
  }, [phoneNumber]);

  const onPhoneNumberPressed = async () => {
    try {
      var phoneNumber = await SmsRetriever.requestPhoneNumber();
      setPhoneNumber(phoneNumber.replace("+91", ""));
    } catch (error) {
      console.log("Error while fetching phoneNumber: ", error.toString());
    }
  };

  useEffect(() => {
    onPhoneNumberPressed();
  }, []);

  const backAction = () => {
    Alert.alert("Hold on!", "Are you sure you want to go back?", [
      { text: "No", onPress: () => null, style: "cancel" },
      { text: "Yes", onPress: () => BackHandler.exitApp() },
    ]);
    return true;
  };

  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", backAction);
    return () =>
      BackHandler.removeEventListener("hardwareBackPress", backAction);
  }, []);

  return (
    <SafeAreaView style={styles.safeContainer}>
      <KeyboardAvoidingWrapper>
        <View>
          <SVGImg style={styles.logo} />
          <Text style={styles.headline}>
            Please enter your mobile number to login:
          </Text>

          <FormInput
            placeholder="Enter mobile number"
            containerStyle={{ marginVertical: 30 }}
            autoCompleteType="tel"
            keyboardType="phone-pad"
            value={phoneNumber}
            onChange={setPhoneNumber}
            autoFocus={true}
            maxLength={10}
            prependComponent={
              <View
                style={{
                  alignItems: "center",
                  justifyContent: "center",
                  borderRightWidth: 1,
                  borderColor: COLORS.gray,
                  marginRight: 10,
                  height: "80%",
                }}
              >
                <Text
                  style={{
                    ...FONTS.h3,
                    color: COLORS.black,
                    paddingRight: 10,
                    // fontWeight: "bold",
                  }}
                >
                  + 91
                </Text>
              </View>
            }
          />

          <Text style={styles.dataUseText}>
            This number will be used for all communication. You shall receive an
            SMS with code for verification. By continuing, you agree to our{" "}
            <Text
              onPress={() => setIsTermsOfUseModalVisible(true)}
              style={styles.termsText}
            >
              Terms of Service
            </Text>{" "}
            &{" "}
            <Text
              onPress={() => setIsPrivacyModalVisible(true)}
              style={styles.termsText}
            >
              Privacy Policy
            </Text>
          </Text>
          <Otp disabled={disabled} type="login" />
        </View>
      </KeyboardAvoidingWrapper>

      {isTermsOfUseModalVisible && (
        <TermsAndPrivacyModal
          isVisible={isTermsOfUseModalVisible}
          setIsVisible={setIsTermsOfUseModalVisible}
          data={termsOfUse}
        />
      )}

      {isPrivacyModalVisible && (
        <TermsAndPrivacyModal
          isVisible={isPrivacyModalVisible}
          setIsVisible={setIsPrivacyModalVisible}
          data={privacyPolicy}
        />
      )}
    </SafeAreaView>
  );
};

export default LoginScreen;
