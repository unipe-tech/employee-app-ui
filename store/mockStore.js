const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
