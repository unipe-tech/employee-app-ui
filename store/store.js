import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";
import aadhaarSlice from "./slices/aadhaarSlice";
import panSlice from "./slices/panSlice";
import bankSlice from "./slices/bankSlice";
import esicSlice from "./slices/esicSlice";
import profileSlice from "./slices/profileSlice";
import navigationSlice from "./slices/navigationSlice";

export const store = configureStore({
  reducer: {
    auth: authSlice,
    aadhaar: aadhaarSlice,
    pan: panSlice,
    bank: bankSlice,
    profile: profileSlice,
    esic: esicSlice,
    navigation: navigationSlice
  },
});
