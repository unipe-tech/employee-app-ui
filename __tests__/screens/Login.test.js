import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { Provider } from "react-redux";
import LoginScreen from "../../screens/00_login/LoginScreen";
import { store } from "../../store/store";
import { TextInput } from "react-native";

let navigation;
beforeEach(() => {
  navigation = {
    dispatch: jest.fn(),
    navigate: jest.fn(),
  };
});

describe("Welcome Screen", () => {
  // snapshot test
  it("should render UI properly", () => {
    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <LoginScreen />
        </NavigationContainer>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // navigates to the next screen => Login Screen
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const component = render(
      <Provider store={store}>
        <NavigationContainer>
          <LoginScreen navigation={{ navigate }} />
        </NavigationContainer>
      </Provider>
    );
    fireEvent.changeText(component.findByTestId("login-number"), "8168176767");
    const event = fireEvent.press(component.getByText("Continue"));
    expect(navigate).toHaveBeenCalledWith("Otp");
  });
});
