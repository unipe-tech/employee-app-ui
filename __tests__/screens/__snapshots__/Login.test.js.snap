// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen should render UI properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
    }
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "height": "100%",
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <RCTScrollView>
          <View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <Image
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": 87,
                    "marginLeft": 70,
                    "marginTop": 30,
                    "width": 250,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#230C45",
                    "fontFamily": "Roboto",
                    "fontSize": 16,
                    "marginLeft": 55,
                    "marginTop": 40,
                  }
                }
              >
                Please enter your mobile number to login:
              </Text>
              <Text
                style={
                  Object {
                    "color": "#020614",
                    "fontFamily": "Noto Sans",
                    "fontSize": 14,
                    "marginLeft": 40,
                    "marginTop": 60,
                  }
                }
              >
                Mobile Number
              </Text>
              <TextInput
                autoCompleteType="tel"
                keyboardType="phone-pad"
                maxLength={13}
                onChangeText={[Function]}
                placeholder="9999999999"
                style={
                  Object {
                    "borderBottomWidth": 1,
                    "height": 50,
                    "marginLeft": 40,
                    "width": "80%",
                  }
                }
                testID="login-number"
                textContentType="telephoneNumber"
                value=""
              />
              <Text
                style={
                  Object {
                    "color": "gray",
                    "fontFamily": "Roboto",
                    "fontSize": 14,
                    "marginLeft": 40,
                    "marginRight": 20,
                    "marginTop": 20,
                  }
                }
              >
                This number will be used for all communication. You shall receive an SMS with code for verification. By continuing, you agree to our
                 
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "#4E46F1",
                      "fontWeight": "bold",
                    }
                  }
                >
                  Terms of Service
                </Text>
                 
                &
                 
                <Text
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "#4E46F1",
                      "fontWeight": "bold",
                    }
                  }
                >
                  Privacy Policy
                </Text>
              </Text>
              <View
                collapsable={false}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#4E46F1",
                    "borderBottomEndRadius": 4,
                    "borderBottomStartRadius": 4,
                    "borderRadius": 6,
                    "borderTopEndRadius": 4,
                    "borderTopStartRadius": 4,
                    "fontSize": 20,
                    "height": 60,
                    "marginTop": 20,
                    "padding": 10,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 320,
                  }
                }
                type="solid"
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderBottomEndRadius": 4,
                        "borderBottomStartRadius": 4,
                        "borderTopEndRadius": 4,
                        "borderTopStartRadius": 4,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      Object {
                        "disabled": true,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onLayout={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 36,
                          "justifyContent": "center",
                          "minWidth": 64,
                          "paddingEnd": 16,
                          "paddingStart": 16,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                          },
                        ]
                      }
                    />
                    <Text
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 1.25,
                            "textTransform": "uppercase",
                          },
                          Object {
                            "color": "#000000",
                          },
                          Array [
                            Object {
                              "color": "#a6a6a6",
                            },
                            Object {
                              "opacity": 1,
                              "textTransform": "none",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Continue
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
